apisix/plugins/mocking.lua

```diff
@@ -229,6 +229,7 @@ function _M.access(conf, ctx)
 
     if conf.response_headers then
         for key, value in pairs(conf.response_headers) do
+            value = core.utils.resolve_var(value, ctx.var)
```

Comment:
When parsed to nil, I think this request header can be ignored

Reply:
If one of the `response_header` = `Foo: bar`. But `bar` is not an inbuilt variable. Thus, `resolve_var` will return nil. In this case we should add the header "as is".

Reply:
I agree with @smileby .
I have reviewed all the codes, `resolve_var` never check nil, and it can work without error.
when the `value` is nil, this header will be ignored as @smileby say.
It work the same as nginx `add_header`, I think it is ok

Reply:
why ignore the header if the value is nil?

Reply:
Our competing product is nginx. I refer to nginx's habits. If nginx `add_header foo "";` is configured, this response header will be ignored and not displayed. This habit may have been accepted by most people.

Reply:
okay my bad, `resolve_var` returns nil when the regex has syntax errors. I was in the notion that it returns nil when the input isn't a built in variable.